btr-api/sample_data/generate_data.py

```diff
@@ -0,0 +1,228 @@
+# limitations under the License.
+"""The BTR data generator script."""
+import sys
+
+import pycountry
+from faker import Faker
+from flask import current_app
+from random import sample, randint
+
+from btr_api import create_app
+from btr_api.models import User, db
+from btr_api.services import SubmissionService
+
+fake = Faker()
+
+INTEREST_TYPES = ['otherInfluenceOrControl', 'shareholding', 'votingRights']
+
+# 1) the locale used for generating sample data, 
+# 2) the number of filings to generate, and
+# 3) the number of SI individuals in a filing
+SETTING = {
+    'en_CA': [10, 3],
+    'fr_FR': [2, 2],
+    'jp_JP': [1, 1]
+}
+
+def get_fake_date() -> str:
+    """Return a fake date."""
+    return str(fake.date_this_decade())
+
+
+def _get_ooc_interests(date):
+    """Return the ooc interests."""
+    ooc_interests: list[dict] = []
+    interest_types = sample(INTEREST_TYPES, randint(1, len(INTEREST_TYPES)))
+    for interest_type in interest_types:
+        interest = {
+            'type': interest_type,
+            'beneficialOwnershipOrControl': True,
+            'directOrIndirect': fake.random_element(['direct', 'indirect', 'unknown']),
+            'startDate': date,
+        }
+        if interest['type'] in ['shareholding', 'votingRights']:
+            interest['share'] = { 'exact': fake.random_int(25, 100) }
+            interest['details'] = 'controlType.sharesOrVotes.registeredOwner'
+        else:
+            interest['details'] = 'controlType.other'
+
+        ooc_interests.append(interest)
+
+    return ooc_interests
+
+
+def _get_ooc_stmnts(entity_stmnt, person_stmnts):
+    """Return the ooc statements."""
+    ooc_stmnts: list[dict] = []
+    for person in person_stmnts:
+        ooc_stmnts.append({
+            'statementID': 'sample-ooc-' + fake.uuid4(),
+            'statementType': 'ownershipOrControlStatement',
+            'statementDate': entity_stmnt['statementDate'],
+            'isComponent': fake.random_element([True, False]),
+            'subject': {'describedByEntityStatement': entity_stmnt['statementID']},
+            'interestedParty': {'describedByPersonStatement': person['statementID']},
+            'publicationDetails': entity_stmnt['publicationDetails'],
+            'interests': _get_ooc_interests(entity_stmnt['statementDate'])
+        })
+
+    return ooc_stmnts
+
+
+def _get_entity_stmnt(statement_date: str):
+    """Return the entity statements."""
+    return {
+        'statementID': 'sample-entity-' + fake.uuid4(),
+        'statementType': fake.random_element(['business', 'branch']),
+        'statementDate': statement_date, 
+        'isComponent': fake.random_element([True, False]),
+        'entityType': 'legalEntity',
+        'name': fake.company(),
+        'identifiers': [{'id': f"TST-E{fake.uuid4()}", 'scheme': 'TST-E'}],
+        'publicationDetails': {
+            "publicationDate":  statement_date,
+            "bodsVersion": "1.0",
```

Comment:
If we want to be closer to real state, we could use what is current revision of BODS. It is v0.3.